cordova-lib/src/cordova/platform.js

```diff
@@ -122,6 +122,12 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                         if (platformAlreadyAdded) {
                             throw new CordovaError('Platform ' + platform + ' already added.');
                         }
+
+                        // Remove the <platform>.json file from the plugins directory, so we start clean (otherwise we
+                        // can get into trouble not installing plugins if someone deletes the platform folder but
+                        // <platform>.json still exists).
+                        removePlatformPluginsJson(projectRoot, target);
```

Comment:
I think this function call should be moved to https://github.com/MSOpenTech/cordova-lib/blob/CB-9051/cordova-lib/src/cordova/platform.js#L174, because the 'addHelper' function is used both by the 'cordova platform add' and 'cordova platform update'. and it is our intention to only do this file deletion when we are performing an 'add' operation.
Another reason might be that we should be doing this file deletion close to the 'installPluginsForTheNewPlatform' function call, to minimize the risk of getting in an inconsistent state where the file has already been deleted even though the 'add' operation ends up throwing an exception down the line.